src/settings/SettingsPage.js

```diff
@@ -2,23 +2,43 @@ import React from 'react'
 import { Helmet } from 'react-helmet'
 import { connect } from 'redux-bundler-react'
 import { translate, Trans } from 'react-i18next'
+import i18n from '../i18n'
+
+// Utils
+import isoLangs from './utils/iso-langs.json'
+
+// Components
+import Tick from '../icons/GlyphSmallTick'
 import Box from '../components/box/Box'
 import Button from '../components/button/Button'
 import JsonEditor from './editor/JsonEditor'
-import Tick from '../icons/GlyphSmallTick'
+import Overlay from '../components/overlay/Overlay'
+import LanguageModal from './language-modal/LanguageModal'
 
 const PAUSE_AFTER_SAVE_MS = 3000
 
 const SettingsPage = ({
   t, tReady,
   isConfigBlocked, isLoading, isSaving,
   hasSaveFailed, hasSaveSucceded, hasErrors, hasLocalChanges, hasExternalChanges,
-  config, onChange, onReset, onSave, editorKey
+  config, onChange, onReset, onSave, editorKey, showEditModal, onLanguageEditOpen, onLanguageEditClose
 }) => (
   <div data-id='SettingsPage'>
     <Helmet>
       <title>{t('title')} - IPFS</title>
     </Helmet>
+
+    <Box className='mb3 pa4'>
+      <div className='flex justify-between items-center'>
+        <p className='ma0 lh-copy charcoal-muted f6'>
+          {t('language', { language: isoLangs[i18n.language].nativeName })}
```

Comment:
`isoLangs[i18n.language].nativeName` is risky. It assumes that whatever comes back from i18n.language will definitly be a key the isoLangs object. 

As an example, `i18n.language` reports `en-GB` on my machine becuase I am so very british i have a locale for it. As such the settings page blows up for me.

Reply:
😮 gotta figure out a work around for that then.